src/views/Vnc.vue

```diff
@@ -1,80 +1,120 @@
 <template>
   <div>
-    <div id="noVNC_status_bar" class="noVNC_status_normal">
+    <div id="noVNC_status_bar" :class="statusClass">
       <div id="noVNC_status">{{status}}</div>
+      <button class="close_button" @click="close(false)">
+        <span>Close</span>
+        <XIcon class="w-4 h-4 ml-1" />
+      </button>
     </div>
-    <div id="view"></div>
+    <div id="view" ref="terminal"></div>
   </div>
 </template>
 
 <script>
+/* global process */
+
+import * as api from '@/account-utils/index'
 import RFB from '@novnc/novnc/core/rfb'
-import { mapGetters } from 'vuex'
+import { XIcon } from '@heroicons/vue/outline'
+import { mapState } from 'vuex'
 
 export default {
   name: 'VNC',
-  title() {
-    return 'VNC'
-  },
-  computed: {
-    ...mapGetters({
-      vncSettings: 'StateVncSettings'
-    })
-  },
+  props: ['server'],
+  components: { XIcon },
   data() {
     return {
       desktopName: '',
       rfb: null,
-      server: null,
-      status: 'Loading...'
+      status: 'Loading...',
+      statusClass: 'noVNC_status_normal'
+
+    }
+  },
+  computed: {
+    ...mapState(['session']),
+    serverId() {
+      return this.$route.params.id
     }
   },
   methods: {
+    close(hasError) {
+      this.$emit('close-console', hasError)
+    },
     connected() {
-      this.status = `Connected to ${this.desktopName}`
+      this.statusClass = 'noVNC_status_normal'
+      this.status = `Connected to ${this.server.settings.name}`
       this.rfb.focus()
     },
+    disconnected() {
+      setTimeout(() => {
+        this.statusClass = 'noVNC_status_error'
+        this.status = `Failed to connect to ${this.server.settings.name}`
+        this.close(true)
+      }, 1000)
```

Comment:
1000 ms may be a bit too fine. I suggest a slightly longer timeout, perhaps 5000 ms.

Reply:
+1